google/cloud/firestore_v1/_helpers.py

```diff
@@ -397,8 +404,12 @@ def decode_dict(value_fields, client) -> dict:
         of native Python values converted from the ``value_fields``.
     """
     value_fields_pb = getattr(value_fields, "_pb", value_fields)
+    res = {key: decode_value(value, client) for key, value in value_fields_pb.items()}
+
+    if res.get("__type__", None) == "__vector__":
+        return Vector(res["value"])
```

Comment:
nit: a comment would be good here. Explain why this is a special case, and what you're doing here

Reply:
Done.